injector/network_disruption.go

```diff
@@ -479,16 +491,25 @@ func (i *networkDisruptionInjector) addServiceFilters(serviceName string, filter
 	return builtServices, nil
 }
 
-// removeServiceFilter delete tc filters using its priority
+// removeServiceFilter delete tc filters for a k8s service
 func (i *networkDisruptionInjector) removeServiceFilter(interfaces []string, tcFilter tcServiceFilter) error {
+	if err := i.removeTcFilter(interfaces, tcFilter.priority); err != nil {
+		return err
+	}
+
+	i.config.Log.Infow(fmt.Sprintf("deleted a tc filter for service %s with priority %d", tcFilter.service, tcFilter.priority), "interfaces", interfaces)
+
+	return nil
+}
+
+// delete tc filters using only its priority
+func (i *networkDisruptionInjector) removeTcFilter(interfaces []string, priority uint32) error {
 	for _, iface := range interfaces {
-		if err := i.config.TrafficController.DeleteFilter(iface, tcFilter.priority); err != nil {
+		if err := i.config.TrafficController.DeleteFilter(iface, priority); err != nil {
```

Comment:
Should we halt early and do not try to remove other filter when we are failing to remove one of them?

Reply:
We are halting early if any error. We could choose not to, just to remove everything we can? Probably safer to just halt on unexpected errors